### PR TITLE
[FEATURE][needs-docs]Add map tool: rectangle 3 points (projected)

### PR DESCRIPTION
**Pull request by lbartoletti**
## Description
I have already added the creation of a rectangle by 3 points where the rectangle takes a length between the first and second point and the second length is equal to the distance between the 3rd and 2nd point.

This new method allows to create a rectangle by 3 points where the second length is the distance between the point and the point projected perpendicularly on the segment (or its extension).

rectangle3pointsprojected

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [X] Commit messages are descriptive and explain the rationale for changes
- [ ] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [X] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [X] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and contain sufficient information in the commit message to be documented
- [X] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [X] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [X] New unit tests have been added for core changes
- [X] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTING.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: src/app/qgsmaptoolrectangle3points.cpp [hunk 1]
```diff
@@ -77,7 +78,11 @@ void QgsMapToolRectangle3Points::cadCanvasMoveEvent( QgsMapMouseEvent *e )
       break;
       case 2:
       {
-        setDistance2( mPoints.at( 1 ).distance( point ) );
+        if ( mCreateMode == DistanceMode )
```
Comment (nyalldawson) [inline src/app/qgsmaptoolrectangle3points.cpp]: use a switch here instead

**Review by nyalldawson — COMMENTED** [review]
Nice feature!

File: src/app/qgsmaptoolrectangle3points.cpp (hunk 1 above)
Comment (lbartoletti) [reply src/app/qgsmaptoolrectangle3points.cpp]: Done. Thanks.

**Comment by nyalldawson** [comment]
@3nids?

**Review by 3nids — APPROVED** [review]
seems good to me!

**Comment by 3nids** [comment]
I do 😀

**Pull request merged**